src/app/view/components/dot-listing-data-table/dot-listing-data-table.component.spec.ts

```diff
@@ -449,13 +452,25 @@ describe('DotListingDataTableComponent', () => {
         });
     });
 
-    it('renders the dot empty state component if items array is empty', fakeAsync(() => {
+    it('should renders the dot empty state component if items array is empty', fakeAsync(() => {
         setRequestSpy([]);
         hostFixture.detectChanges();
         tick(1);
         hostFixture.detectChanges();
-        const empty = de.query(By.css('dot-empty-state'));
-        expect(empty.nativeElement.innerText).toBe('Im empty');
+        const emptyState = de.query(By.css('dot-empty-state'));
+        expect(emptyState.nativeElement.innerText).toBe('Im empty');
+    }));
+
+    it('should show no results message if filtered content is empty', fakeAsync(() => {
+        setRequestSpy([]);
+        hostFixture.detectChanges();
+        tick(1);
+        comp.globalSearch.nativeElement.value = 'test';
+        comp.globalSearch.nativeElement.dispatchEvent(new Event('input'));
+        tick(de.componentInstance.filterDelay + 1);
```

Comment:
because of this, https://github.com/primefaces/primeng/blob/master/src/app/components/table/table.ts#L201